### ISSUE TITLE
Emulator speed bug

### ISSUE THREAD

**Issue by GhbSmwc** [devinacker/bsnes-plus#169]
During my testing with smw ROM hacking, I've found a few glitches regarding the emulator speed:

-If you fast-foward (by pressing the &ldquo;`&rdquo; (the tilde character on the top-left corner of the keyboard just down from ESC)), then close the window, the emulator maintains this state when re-opened. If you fast-foward, it actually temporally change the speed setting to &ldquo;Fastest&rdquo; and turns off (unchecks) &ldquo;Sync Video&rdquo; and &ldquo;Sync Audio&rdquo;. When closed and re-opened, it doesn't restore the settings.

Another thing: When having video and audio syncs on, and change the speed above normal (not using the fast-foward button), the audio and video runs the exact same speed, just the audio is higher pitched. You might want to fix that.

**Comment by GhbSmwc** [comment]
Also, there is another issue, sometimes, the game runs slow when the syncs (both video and audio) are on. When both turned off and you changed the speeds, the game's speed is the same with only the audio pitch changes.

**Comment by GhbSmwc** [comment]
Not sure if the emulator's speed setting on the UI and the actual speed are seperate and have a bug where both of their states aren't the same (like it shows that the speed is &ldquo;fastest&rdquo;, but is actually running on &ldquo;slowest&rdquo;)

**Comment by devinacker** [comment]
>Another thing: When having video and audio syncs on, and change the speed above normal (not using the fast-foward button), the audio and video runs the exact same speed, just the audio is higher pitched. You might want to fix that.

"sync video" and "sync audio" are designed to make sure that emulation does not continue until video and/or audio have finished updating - in other words, prevents emulation from running too fast for the video or audio output to keep up. This is why fast-forwarding temporarily disables both of these settings.

(Ideally, fast-forwarding should just cause the emulator to temporarily ignore the current speed/sync settings to prevent these UI problems from happening; it's something I've also had to fix problems with in the past and I'll probably try to do something better about it in the v4 release)

Generally, having audio and video sync enabled at the same time will cause some slowdown because you're attempting to force emulation to wait on two different things that are both running independently of each other. Instead, it is recommended to either:
* enable audio sync, disable video sync - provides smooth audio, but sometimes somewhat choppy video
* enable video sync, disable audio sync - provides smooth video, but with audio stuttering/crackling

The latter is generally the worse option, which is one reason why higan doesn't even have a video sync option anymore.
